src/java.base/macosx/classes/sun/nio/fs/BsdNativeDispatcher.java

```diff
@@ -82,26 +82,24 @@ private static native int clonefile0(long srcAddress, long dstAddress,
                                          int flags);
 
     /**
-     * setattrlist(const char* path, struct attrlist* attrList, void* attrBuf,
-     *             size_t attrBufSize, unsigned long options)
+     * fsetattrlist(int fd, struct attrlist* attrList, void* attrBuf,
+     *              size_t attrBufSize, unsigned long options)
      */
-    static void setattrlist(UnixPath path, int commonattr, long modTime,
-                            long accTime, long createTime, long options)
+    static void fsetattrlist(int fd, int commonattr, long modTime,
+                             long accTime, long createTime, long options)
         throws UnixException
     {
-        try (NativeBuffer buffer = copyToNativeBuffer(path)) {
-            long comp = Blocker.begin();
-            try {
-                setattrlist0(buffer.address(), commonattr, modTime, accTime,
-                             createTime, options);
-            } finally {
-                Blocker.end(comp);
+        long comp = Blocker.begin();
+        try {
+            fsetattrlist0(fd, commonattr, modTime, accTime,
+                          createTime, options);
+        } finally {
+            Blocker.end(comp);
             }
```

Comment:
Can you check the formatting here, the end brace seems to be in pushed out.

Reply:
Corrected in BsdFileAttributeViews-setTimes-8298187.